micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -92,10 +112,10 @@ public void stop() {
 
     @Override
     public void close() {
+        stop();
```

Comment:
I like shutting down the executor first.

Reply:
Yeah, I think it always should have been this way, regardless of this issue. No one ever complained, and no tests exist that would catch it being a problem, but it logically makes sense to do this first. I noticed it when working on the StepMeterRegistry's close method, but didn't end up opening an issue for it. Glad we've got it updated now. I didn't find any history on why it intentionally would have been the way it was before.